compose/cli/main.py

```diff
@@ -415,7 +415,7 @@ def exec_command(self, options):
         """
         index = int(options.get('--index'))
         service = self.project.get_service(options['SERVICE'])
-        detach = options['-d']
+        detach = options.get('-d') or options.get('--detach')
```

Comment:
You can shorten this to `options.get('--detach')`, which will be set even if the user uses the short flag.